product/live_metrics/middleware_server.yaml

```diff
@@ -12,26 +12,27 @@ included_children:
 
 supported_metrics:
-- WildFly Memory Metrics~Heap Used: mw_heap_used
-- WildFly Memory Metrics~Heap Max: mw_heap_max
-- WildFly Memory Metrics~Heap Committed: mw_heap_committed
-- WildFly Memory Metrics~NonHeap Used: mw_non_heap_used
-- WildFly Memory Metrics~NonHeap Committed: mw_non_heap_committed
-- WildFly Memory Metrics~Accumulated GC Duration: mw_accumulated_gc_duration
-- WildFly Aggregated Web Metrics~Aggregated Servlet Request Time: mw_aggregated_servlet_time
-- WildFly Aggregated Web Metrics~Aggregated Servlet Request Count: mw_aggregated_servlet_request_count
-- WildFly Aggregated Web Metrics~Aggregated Active Web Sessions: mw_aggregated_active_web_sessions
-- WildFly Aggregated Web Metrics~Aggregated Rejected Web Sessions: mw_aggregated_rejected_web_sessions
-- WildFly Aggregated Web Metrics~Aggregated Expired Web Sessions: mw_aggregated_expired_web_sessions
-- WildFly Aggregated Web Metrics~Aggregated Max Active Web Sessions: mw_aggregated_max_active_web_sessions
-- WildFly Threading Metrics~Thread Count: mw_thread_count
-- Server Availability~App Server: mw_availability_app_server
-- Transactions Metrics~Number of Aborted Transactions: mw_tx_aborted
-- Transactions Metrics~Number of In-Flight Transactions: mw_tx_inflight
-- Transactions Metrics~Number of Committed Transactions: mw_tx_committed
-- Transactions Metrics~Number of Transactions: mw_tx_total
-- Transactions Metrics~Number of Application Rollbacks: mw_tx_application_rollbacks
-- Transactions Metrics~Number of Resource Rollbacks: mw_tx_resource_rollbacks
-- Transactions Metrics~Number of Timed Out Transactions: mw_tx_timeout
-- Transactions Metrics~Number of Nested Transactions: mw_tx_nested
-- Transactions Metrics~Number of Heuristics: mw_tx_heuristics
+  default:
+  - Heap Used: mw_heap_used
+  - Heap Max: mw_heap_max
+  - Heap Committed: mw_heap_committed
+  - NonHeap Used: mw_non_heap_used
+  - NonHeap Committed: mw_non_heap_committed
+  - Accumulated GC Duration: mw_accumulated_gc_duration
+  - Aggregated Servlet Request Time: mw_aggregated_servlet_time
+  - Aggregated Servlet Request Count: mw_aggregated_servlet_request_count
+  - Aggregated Active Web Sessions: mw_aggregated_active_web_sessions
+  - Aggregated Rejected Web Sessions: mw_aggregated_rejected_web_sessions
+  - Aggregated Expired Web Sessions: mw_aggregated_expired_web_sessions
+  - Aggregated Max Active Web Sessions: mw_aggregated_max_active_web_sessions
+  - Thread Count: mw_thread_count
+  - App Server: mw_availability_app_server
+  - Number of Aborted Transactions: mw_tx_aborted
+  - Number of In-Flight Transactions: mw_tx_inflight
+  - Number of Committed Transactions: mw_tx_committed
+  - Number of Transactions: mw_tx_total
+  - Number of Application Rollbacks: mw_tx_application_rollbacks
+  - Number of Resource Rollbacks: mw_tx_resource_rollbacks
+  - Number of Timed Out Transactions: mw_tx_timeout
+  - Number of Nested Transactions: mw_tx_nested
+  - Number of Heuristics: mw_tx_heuristics
```

Comment:
How to avoid collisions here? This metric is part of `Transaction Manager`, what happens if e.g. MiddlewareServer also have a `Number of Heuristics` ?

Reply:
Then we will need to change that collision in the inventory side.
This is the only corner case when we need to bring children metrics into the parent one.
If more cases appear perhaps we should consider to remap and add proper entities also in MiQ.
But I guess that for this case it will be treated as same was as the original code.

Reply:
But even in that case, looking in the implementation, it is safe, as long as child/parent has a different miq metric name.
Number of Heuristics: mw_tx_heuristics
...
Number of Heuristics: mw_tx_heuristics_parent

Or similar, as we know that within the resource we don't have such collision in the inventory.

Reply:
I think the problem is the other way around, there is no way to say: use `Number of Heuristics` from the `parent` or use `Number of Heuristics` from this (e.g. `Transaction Manager`) children.

But I agree that we could ensure that we don't have any collision on inventory side until is really needed to handle this case.